dali_tf_plugin/dali_dataset_op.cc

```diff
@@ -311,34 +455,18 @@ REGISTER_OP("DALIDataset")
   .Attr("prefetch_queue_depth: int")
   .Attr("cpu_prefetch_queue_depth: int")
   .Attr("gpu_prefetch_queue_depth: int")
-  .Attr("shapes: list(shape) >= 1")
-  .Attr("dtypes: list({half, float, uint8, int16, int32, int64}) >= 1")
+  .Attr("output_shapes: list(shape) >= 1")
+  .Attr("output_dtypes: list({bool, half, float, uint8, uint16, uint32, uint64, int8, int16, int32, int64}) >= 1")
   .Output("handle: variant")
   .SetIsStateful()
-  .SetShapeFn([](shape_inference::InferenceContext* c) {
-    std::vector<PartialTensorShape> shapes;
-    TF_RETURN_IF_ERROR(c->GetAttr("shapes", &shapes));
-    c->ExpandOutputs(shapes.size());
-
-    for (unsigned i = 0; i < shapes.size(); ++i) {
-      if (shapes[i].dims() > 0) {
-        shape_inference::ShapeHandle passed_shape;
-        TF_RETURN_IF_ERROR(
-            c->MakeShapeFromPartialTensorShape(shapes[0], &passed_shape));
-        TF_RETURN_IF_ERROR(
-            c->WithRank(passed_shape, shapes[0].dims(), &passed_shape));
-        c->set_output(i, passed_shape);
-      }
-    }
-    return Status::OK();
-  })
+  .SetShapeFn(shape_inference::ScalarShape)
```

Comment:
a comment here would help to understand (why are we setting to ScalarShape?)

Reply:
Because all the TF datasets are doing the same. I can add a comment, it's because the dataset is basically returning a opaque pointer to a Tensor instead of the Tensor itself.